module/VuFind/src/VuFind/View/Helper/Root/ProxyUrl.php

```diff
@@ -66,10 +80,68 @@ public function __construct($config = null)
      */
     public function __invoke($url)
     {
-        $usePrefix = !isset($this->config->EZproxy->prefixLinks)
-            || $this->config->EZproxy->prefixLinks;
+        $useWebService = $this->config->EZproxy->prefixLinksWebServiceUrl ?? false;
+        if ($useWebService) {
+            $usePrefix = $this->checkUrl($url) ?? $this->checkConfig();
+        } else {
+            $usePrefix = $this->checkConfig();
+        }
+
         return ($usePrefix && isset($this->config->EZproxy->host))
             ? $this->config->EZproxy->host . '/login?qurl=' . urlencode($url)
             : $url;
     }
+
+    /**
+     * Return the configured prefixLinks setting.
+     *
+     * @return bool The configured setting, or the default
+     */
+    protected function checkConfig()
+    {
+        return $this->config->EZproxy->prefixLinks ?? true;
+    }
+
+    /**
+     * Check whether the given URL requires the proxy prefix.  Cache the response.
+     *
+     * @param string $url The raw URL to check
+     *
+     * @return mixed Whether the URL should be prefixed, or null if it can't be determined
+     */
+    protected function checkUrl($url)
+    {
+        $domain = parse_url($url, PHP_URL_SCHEME)
+         . '://'
+         . parse_url($url, PHP_URL_HOST);
+        $cacheKey = "proxyUrl-domainToUsePrefix-$domain";
+        $usePrefix = $this->getCachedData($cacheKey);
+        if (null === $usePrefix) {
+            $usePrefix = $this->queryWebService($domain);
+            $this->putCachedData($cacheKey, $usePrefix);
+        }
+        return $usePrefix;
+    }
+
+    /**
+     * Query the web service on whether to prefix URLs to a given domain.
+     *
+     * @param $domain The domain
+     *
+     * @return mixed Whether the URL should be prefixed, or null if it can't be determined
+     */
+    protected function queryWebService($domain)
+    {
+        $prefixLinksWebServiceUrl = $this->config->EZproxy->prefixLinksWebServiceUrl;
+        $queryUrl = $prefixLinksWebServiceUrl . '?url=' . urlencode($domain);
+        try {
+            $client = $this->httpService->createClient($queryUrl);
```

Comment:
Since you're using the httpService, I believe that you could simplify several lines of this code into:

```
$response = $this->httpService->get($prefixLinksWebServiceUrl, ['url' => $domain]);
```

(See the [get method](https://github.com/vufind-org/vufindhttp/blob/dev/src/VuFindHttp/HttpService.php#L197) here if you want all the details of what it's doing).

Reply:
Nice, updated to use get().  Wherever I looked for httpService in the code wasn't doing that.